src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+                                                                                                                                                                                              region,
+                                                                                                                                                                                              resourceGroup,
+                                                                                                                                                                                              networkInterfaceName,
+                                                                                                                                                                                              creatableVirtualNetwork,
+                                                                                                                                                                                              creatableNetworkSecurityGroup,
+                                                                                                                                                                                              creatablePublicIPAddress);
+
+                                                                                return prepareVirtualMachine(instance,
+                                                                                                             azureService,
+                                                                                                             resourceGroup,
+                                                                                                             region,
+                                                                                                             createUniqInstanceTag(instanceTag,
+                                                                                                                                   instanceNumber),
+                                                                                                             image,
+                                                                                                             creatableNetworkInterface);
+                                                                            })
+                                                                            .collect(Collectors.toList());
+
+        // Create all VMs in parallel and collect IDs
+        return azureService.virtualMachines()
+                           .create(creatableVirtualMachines)
+                           .values()
+                           .stream()
+                           .map(vm -> instance.withTag(vm.name()).withId(vm.vmId()).withNumber("1"))
```

Comment:
Yeah this is something weird. When we create an instance we provide an Instance object in argument which contains the 'number' field that specify how much instances we need to create.
Then after creating the desired instances, we return a list of single instance and the 'number' field need therefore to be set to "1".
Where is the documentation you mentioned?